website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,438 @@
+*   Config files: `.graphqlrc`, `codegen.yml`, etc.
+
+Changing anything in the Maintainers' domain should notify them. Unfortunately, this happens regularly in normal workflows. Maintainers also have their team's work and general package and security updates to worry about. So, being notified on unrelated PRs quickly lead to burnout.
+
+For example, if a new resolver is added, they must be manually added to the resolver map. So, Maintainers are notified on every new resolver. It may look like this if you are using [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server):
+
+```ts
+// server.ts
+import { createServer } from 'node:http'
+import { createSchema, createYoga } from 'graphql-yoga'
+import { Auth } from "./schema/auth/resolvers";
+import { book, Book } from "./schema/auth/resolvers";
+import { user, User } from "./schema/user/resolvers";
+
+const schema = createSchema({
+  typeDefs: `...`,
+  resolvers: { // This is the resolver map
+    Query: {
+      user,
+      book,
+    },
+    Auth,
+    Book
+    User,
+  }
+});
+
+const yoga = createYoga({ schema });
+const server = createServer(yoga)
+server.listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+})
+```
+
+(TODO: not sure if we are ok with calling out the caveats of mergeResolvers from @graphql-tools/merge here)
+
+There is a way to mitigate this issue: each schema module exports an object of resolvers and they are merged with [mergeResolvers from @graphql-tools/merge](https://the-guild.dev/graphql/tools/docs/schema-merging#merging-resolvers). This means the Maintainers are notified on every new module instead of every resolver:
+
+```ts
+// server.ts
+import { createServer } from 'node:http'
+import { createSchema, createYoga } from 'graphql-yoga'
+import { mergeResolvers } from '@graphql-tools/merge';
+import * as authResolvers from "./schema/auth/resolvers";
+import * as bookResolvers from "./schema/book/resolvers";
+import * as userResolvers from "./schema/user/resolvers";
+
+const schema = createSchema({
+  typeDefs: `...`,
+  resolvers: mergeResolvers([ // Resolver map is the result of merging module resolvers
+    authResolvers,
+    bookResolvers,
+    userResolvers,
+  ])
+});
+
+// rest of server config
+```
+
+However, `mergeResolvers` has a runtime and it is simply merging plain JavaScript objects. Thus, there is a risk of someone accidentally overriding others' resolvers. The overriding issue is hard to find in large codebases with hundreds of resolvers.
+
+<img alt="mergeResolvers can cause conflicts and pain" src="/blog-assets/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset/merge-resolvers.gif" style={{width: "80%", margin: "0 auto"}}/>
+
+Another commonly used feature is [mappers](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers#mappers). This feature allows resolvers to return mapper objects instead of GraphQL output types.
```

Comment:
**Question:** Is there a The Guild blog post about GraphQL being more than just forwarding API responses to the client? Or is there a post about how mappers work?

Trying to explain it in this post takes away too much from the main content